crates/networking/p2p/sync_manager.rs

```diff
@@ -0,0 +1,155 @@
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+use ethrex_blockchain::Blockchain;
+use ethrex_common::H256;
+use ethrex_storage::{error::StoreError, Store};
+use tokio::sync::Mutex;
+use tokio_util::sync::CancellationToken;
+use tracing::warn;
+
+use crate::{
+    kademlia::KademliaTable,
+    sync::{SyncMode, Syncer},
+};
+
+pub enum SyncStatus {
+    Active(SyncMode),
+    Inactive,
+}
+
+/// Abstraction to interact with the active sync process without disturbing it
+#[derive(Debug)]
+pub struct SyncManager {
+    /// This is also held by the Syncer and allows tracking it's latest syncmode
+    /// It is a READ_ONLY value, as modifications will disrupt the current active sync progress
+    snap_enabled: Arc<AtomicBool>,
+    syncer: Arc<Mutex<Syncer>>,
+    last_fcu_head: Arc<Mutex<H256>>,
+    store: Store,
+}
+
+impl SyncManager {
+    pub fn new(
+        peer_table: Arc<Mutex<KademliaTable>>,
+        sync_mode: SyncMode,
+        cancel_token: CancellationToken,
+        blockchain: Arc<Blockchain>,
+        store: Store,
+    ) -> Self {
+        let snap_enabled = Arc::new(AtomicBool::new(matches!(sync_mode, SyncMode::Snap)));
+        let syncer = Arc::new(Mutex::new(Syncer::new(
+            peer_table,
+            snap_enabled.clone(),
+            cancel_token,
+            blockchain,
+        )));
+        Self {
+            snap_enabled,
+            syncer,
+            last_fcu_head: Arc::new(Mutex::new(H256::zero())),
+            store,
+        }
+    }
+
+    /// Creates a dummy SyncManager for tests where syncing is not needed
+    /// This should only be used in tests as it won't be able to connect to the p2p network
+    pub fn dummy() -> Self {
+        Self {
+            snap_enabled: Arc::new(AtomicBool::new(false)),
+            syncer: Arc::new(Mutex::new(Syncer::dummy())),
+            last_fcu_head: Arc::new(Mutex::new(H256::zero())),
+            store: Store::new("temp.db", ethrex_storage::EngineType::InMemory)
+                .expect("Failed to create test DB"),
+        }
+    }
```

Comment:
Maybe gate it behind the test feature?
```suggestion
    /// Creates a dummy SyncManager for tests where syncing is not needed
    /// This should only be used in tests as it won't be able to connect to the p2p network
    #[cfg(test)]
    pub fn dummy() -> Self {
        Self {
            snap_enabled: Arc::new(AtomicBool::new(false)),
            syncer: Arc::new(Mutex::new(Syncer::dummy())),
            last_fcu_head: Arc::new(Mutex::new(H256::zero())),
            store: Store::new("temp.db", ethrex_storage::EngineType::InMemory)
                .expect("Failed to create test DB"),
        }
    }
```

That way trying to use it in regular code results in a compilation error.

Reply:
This seems to break tests on the rpc crate